types/webpack/index.d.ts

```diff
@@ -44,13 +44,23 @@ import { ConcatSource } from 'webpack-sources';
 export = webpack;
 
 declare function webpack(
-    options: webpack.Configuration,
-    handler: webpack.Compiler.Handler
+    options:
+        | webpack.Configuration
+        | ((
+              env: string | Record<string, boolean | number | string>,
+              args: Record<string, string>,
+          ) => webpack.Configuration | Promise<webpack.Configuration>),
+    handler: webpack.Compiler.Handler,
 ): webpack.Compiler.Watching | webpack.Compiler;
 declare function webpack(options?: webpack.Configuration): webpack.Compiler;
 
 declare function webpack(
-    options: webpack.Configuration[],
+    options:
+        | webpack.Configuration[]
+        | ((
```

Comment:
Can you declare this somewhere and use it her and in line 49? Naming is up to you but I would name it `ConfigurationFunction`

Reply:
I did before and the linter broke alerting that it can be a union type, the fix was directly pointed by the linter itself

Reply:
Should I declare the whole options as a type? 🤔

Reply:
@rubenspgcavalcante this is in express merge but I haven't merged it because it sounds like you're still considering a change here. Please let me know if that isn't the case. Thanks!

Reply:
For me looks ok @jessetrinity, you can merge :)
thanks in advance!